### DISCUSSION TITLE
Documentation or comments explaining syntax for Settings.cfg, and possible expansion on those features.

### DISCUSSION THREAD

**Discussion by Alpaccalypse** [zer0Kerbal/ExceptionDetector#41] [Ideas]
E.g. Does "DoublePass" and "SinglePass" mean that ED will ignore errors from the listed mods if they happen only twice, or once, respectively? I am assuming this is the case, going on context. Same goes for "HideKnows" which I assume would ignore exceptions that are registered as not genuine errors, like the one claiming not to be an error, which I reported as a bug in logging. These features would be great to expand upon, as I seemed to be getting at least a handful of errors on a completely fresh and unmodded install, just this morning.

Just editing to add: the other reason for expanding upon features for custom exceptions it both how un-curated and outdated a lot of mods for KSP are at this point, and even more significatnly, how many modders report that the hundreds of exceptions being thrown by the mod are perfectly fine, working as intended, and unavoidalbe... It makes checking for "proper?" errors a harrowing experience. ExceptionDetector is already saving me a lot of scrolling and looking for red highlights; it would be even better if I could get it to temporarily ignore the same errors I have chosen to ignore for the moment, when testing things.

**Comment by zer0Kerbal** [comment]
### excellent observations.

I _believe_ that single vs. double pass refers to how EXCD logs the information

_❔possibly if the log entry is one or multiple lines long❓_

from the code:

### Single
`if (!doublePass) singlePass = CheckPass(condition, SinglePassValues);`

### Double

`if (doublePass) // generally high level built-in modules of pattern msg 1: 'loading <name>' msg 2: 'failed message'    <--// note the lack of <name> on second message `

### settings.cfg

needs to be updated/modernized to handle all the new items that are just an annoyance like mentalities.

### Actual live example log from my dev rig

[ed.log](https://github.com/zer0Kerbal/ExceptionDetector/files/11377484/ed.log)

I need to dig in and understand the internal functioning more.

@Lisias

**Comment by Alpaccalypse** [comment]
Thanks for sharing the info and the extensive explanation in response to my bug report. I'm about to try to add approx 100 mods onto a currently entirely clean KSP install. I'll be sure to update you if I stumble upon anything relevant, whilst using EXCD to try and achieve a heavily modded game, with as close to zero error reports as possible.

**Reply by zer0Kerbal** [reply]
welcome. good luck! 🍀🌠🤞
I've done the same - except it has a few more mods and well it is quite possible that you are much more sane than me! 😁
